pkg/testing/ess/provisioner.go

```diff
@@ -62,89 +60,77 @@ func (p *provisioner) SetLogger(l runner.Logger) {
-		// allow up to 2 minutes for each create request
-		createCtx, createCancel := context.WithTimeout(ctx, 2*time.Minute)
-		resp, err := p.createDeployment(createCtx, r,
-			map[string]string{
-				"division":          "engineering",
-				"org":               "ingest",
-				"team":              "elastic-agent",
-				"project":           "elastic-agent",
-				"integration-tests": "true",
-			})
-		createCancel()
-		if err != nil {
-			return nil, err
-		}
-		results[r] = resp
-	}
+// Create creates a stack.
+func (p *provisioner) Create(ctx context.Context, request runner.StackRequest) (runner.Stack, error) {
+	// allow up to 2 minutes for request
+	createCtx, createCancel := context.WithTimeout(ctx, 2*time.Minute)
+	defer createCancel()
+	resp, err := p.createDeployment(createCtx, request,
+		map[string]string{
+			"division":          "engineering",
+			"org":               "ingest",
+			"team":              "elastic-agent",
+			"project":           "elastic-agent",
+			"integration-tests": "true",
+		})
+	if err != nil {
+		return runner.Stack{}, err
+	}
+	return runner.Stack{
+		ID:            request.ID,
+		Version:       request.Version,
+		Elasticsearch: resp.ElasticsearchEndpoint,
+		Kibana:        resp.KibanaEndpoint,
+		Username:      resp.Username,
+		Password:      resp.Password,
+		Internal: map[string]interface{}{
+			"deployment_id": resp.ID,
+		},
+		Ready: false,
+	}, nil
+}
 
-	// set a long timeout
-	// this context travels up to the magefile, clients that want a shorter timeout can set
-	// it via mage's -t flag
-	readyCtx, readyCancel := context.WithTimeout(ctx, 25*time.Minute)
-	defer readyCancel()
-
-	g, gCtx := errgroup.WithContext(readyCtx)
-	for req, resp := range results {
-		g.Go(func(req runner.StackRequest, resp *CreateDeploymentResponse) func() error {
-			return func() error {
-				ready, err := p.client.DeploymentIsReady(gCtx, resp.ID, 30*time.Second)
-				if err != nil {
-					return fmt.Errorf("failed to check for cloud %s to be ready: %w", req.Version, err)
-				}
-				if !ready {
-					return fmt.Errorf("cloud %s never became ready: %w", req.Version, err)
-				}
-				return nil
-			}
-		}(req, resp))
+// WaitForReady should block until the stack is ready or the context is cancelled.
+func (p *provisioner) WaitForReady(ctx context.Context, stack runner.Stack) (runner.Stack, error) {
+	deploymentID, err := p.getDeploymentID(stack)
+	if err != nil {
+		return stack, fmt.Errorf("failed to get deployment ID from the stack: %w", err)
 	}
-	err := g.Wait()
+	// allow up to 10 minutes for it to become ready
+	ctx, cancel := context.WithTimeout(ctx, 10*time.Minute)
+	defer cancel()
+	p.logger.Logf("Waiting for cloud %s to be ready (%s) [id: %s]", stack.Version, stack.ID, deploymentID)
+	ready, err := p.client.DeploymentIsReady(ctx, deploymentID, 30*time.Second)
 	if err != nil {
-		return nil, err
+		return stack, fmt.Errorf("failed to check for cloud %s to be ready: %w", stack.Version, err)
```

Comment:
[Blocker]
Wouldn't be valuable to have the deploymetID as well? In case someone wants to investigate what cause the stack to not be ready.
```suggestion
		return stack, fmt.Errorf("failed to check for cloud stack %s to be ready: %w", stack.Version, err)
```

Reply:
Updated.